src/Network/Wai/Middleware/Auth/OAuth2.hs

```diff
@@ -88,33 +84,28 @@ getClientSecret = id
 getRedirectURI :: U.URIRef a -> S.ByteString
 getRedirectURI = U.serializeURIRef'
 
-getAccessToken :: OA2.OAuth2Token -> S.ByteString
-getAccessToken = encodeUtf8 . OA2.atoken . OA2.accessToken
+encodeAccessToken :: OA2.OAuth2Token -> S.ByteString
+encodeAccessToken = SL.toStrict . encode . OAuth2TokenBinary
+
+newtype OAuth2TokenBinary =
+  OAuth2TokenBinary { unOAuth2TokenBinary :: OA2.OAuth2Token }
+
+instance Binary OAuth2TokenBinary where
+  put (OAuth2TokenBinary token) = do
```

Comment:
Here is a much easier approach, just do what `binary` does when it derives it:
```
put (OAuth2TokenBinary token) = gput . from
get = OAuth2TokenBinary . to <$> gget
```

Haven't tried it, but I can't see why it wouldn't work

Reply:
I was convinced that would work too, but no such luck :). This is the error I get when I try it.

```
    • No instance for (Binary OA2.AccessToken)
        arising from a use of ‘gput’
```

Reply:
Oh yeah, that's right, `OAuth2Token` is full of newtype wrappers :)
It will be good to add those Binary instances upstream. Until then what you've added looks good.

Reply:
I created a PR to the `hoauth2`:

https://github.com/freizl/hoauth2/pull/106

I noticed that the `binary` package is a dependency of a demo executable of that package, but not the library itself, so there's a bit of a downside to adding these instances. We'll have the see how the maintainer responds.

Reply:
I see, I didn't realize that, since I only glanced at dependencies on hackage. Whatever the outcome of that above PR might be it is not gonna be a blocker for us. In either case good job on all of your PRs.